skiko/src/awtTest/kotlin/org/jetbrains/skiko/SkiaFontProviderTest.kt

```diff
@@ -0,0 +1,107 @@
+package org.jetbrains.skiko
+
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.test.runTest
+import org.jetbrains.skiko.context.isRunningOnJetBrainsRuntime
+import org.jetbrains.skiko.util.assertOpensAreSet
+import org.junit.Assume
+import org.junit.Test
+import kotlin.test.assertContains
+import kotlin.test.assertTrue
+
+
+@OptIn(ExperimentalCoroutinesApi::class)
+class SkiaFontProviderTest {
+    private val provider = FontProvider.Skia
+
+    @Test
+    fun `should contain at least some font families`() = runTest {
+        val families = provider.familyNames()
+        assertTrue(families.isNotEmpty(), "Available font families must not be empty")
+    }
+
+    @Test
+    fun `should include 'System Font' and its AWT alias in system font families (only on macOS)`() = runTest {
+        Assume.assumeTrue(hostOs == OS.MacOS)
+
+        val systemFamilies = provider.familyNames()
+        assertContains(
+            iterable = systemFamilies,
+            element = FontFamilyKey.Apple.AppleSystemUiFont.familyName,
+            message = ".AppleSystemUIFont not found in system families"
+        )
+        assertContains(
+            iterable = systemFamilies,
+            element = FontFamilyKey.Apple.SystemFont.familyName,
+            message = "System Font not found in system families"
+        )
+    }
+
+    @Test
+    fun `should be able to resolve logical fonts when running on JetBrains Runtime`() = runTest {
+        Assume.assumeTrue(isRunningOnJetBrainsRuntime())
+
+        assertOpensAreSet()
+        val logicalFamilies = AwtFontUtils.fontFamilyNamesOrNull()!!
+            .onlyAwtLogicalFamilies()
+            .toSet()
+
+        val missingLogicalFamilies = mutableSetOf<String>()
+        for (logicalFamily in logicalFamilies) {
+            val resolved = AwtFontUtils.resolvePhysicalFontNameOrNull(logicalFamily)
+            if (resolved == null) {
+                missingLogicalFamilies += logicalFamily
+            }
+        }
+
+        assertTrue(
+            actual = missingLogicalFamilies.isEmpty(),
+            message = "These logical font families can't be resolved:\n" +
+                    missingLogicalFamilies.joinToString("\n") { " * $it" }
+        )
+    }
+
+    @Test
+    fun `should provide all the system fonts also available via AWT`() = runTest {
```

Comment:
It requires a lot of rewriting, so we can just keep in mind that Compose panels won't support variable fonts (maybe only some of them). But that can be a blocker for production ready functionality.

Reply:
I'd punt variable fonts support for a subsequent PR — we need to get the basic support shipped ASAP. This may be a blocker down the road but isn't yet for the day 1 scenarios we're supporting, and this PR is already a huge step forward from the previous situation.

Nikolay said he was going to take on further changes to the font loading code, I can work with him to make those changes. What do you think?

Reply:
> On Windows/Linux we can retrieve the file name, and load it in Skia.

On Windows we have to get the native pointer too with this approach :(

Both "Cambria" and "Cambria Math" variant point to "C:\Windows\Fonts\cambria.ttc"

Reply:
Ah that's because that is a composite font (`.ttc` file). It's something we can't do too much about, as they are only kinda-sorta supported by AWT. What AWT does when it loads ttc files is it goes "oh, cool, get me all the fonts in this file" and then does `fonts[0]`, with a nice TODO saying essentially "huh we may wanna revisit this"